drivers/kscan/kscan_mchp_xec_v2.c

```diff
@@ -0,0 +1,454 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_xec_kscan_v2
+
+#include <errno.h>
+#include <device.h>
+#include <drivers/kscan.h>
+#include <drivers/interrupt_controller/intc_mchp_xec_ecia.h>
+#include <kernel.h>
+#include <soc.h>
+#include <sys/atomic.h>
+#include <logging/log.h>
+
+#define LOG_LEVEL CONFIG_KSCAN_LOG_LEVEL
+LOG_MODULE_REGISTER(kscan_mchp_xec);
+
+#define MAX_MATRIX_KEY_COLS CONFIG_KSCAN_XEC_COLUMN_SIZE
+#define MAX_MATRIX_KEY_ROWS CONFIG_KSCAN_XEC_ROW_SIZE
+
+#define KEYBOARD_COLUMN_DRIVE_ALL       -2
+#define KEYBOARD_COLUMN_DRIVE_NONE      -1
+
+/* Poll period/debouncing rely onthe 32KHz clock with 30 usec clock cycles */
+#define CLOCK_32K_HW_CYCLES_TO_US(X) \
+	(uint32_t)((((uint64_t)(X) * 1000000U) / sys_clock_hw_cycles_per_sec()))
+/* Milliseconds in microseconds */
+#define MSEC_PER_MS 1000U
+/* Number of tracked scan times */
+#define SCAN_OCURRENCES 30U
+/* Thread stack size */
+#define TASK_STACK_SIZE 1024
+
+struct kscan_xec_config {
+	uint32_t base_addr;
+	uint8_t girq;
+	uint8_t girq_pos;
+	uint8_t pcr_idx;
+	uint8_t pcr_bitpos;
+};
+
+struct kscan_xec_data {
+	/* variables in usec units */
+	uint32_t deb_time_press;
+	uint32_t deb_time_rel;
+	int64_t poll_timeout;
+	uint32_t poll_period;
+	uint8_t matrix_stable_state[MAX_MATRIX_KEY_COLS];
+	uint8_t matrix_unstable_state[MAX_MATRIX_KEY_COLS];
+	uint8_t matrix_previous_state[MAX_MATRIX_KEY_COLS];
+	/* Index in to the scan_clock_cycle to indicate start of debouncing */
+	uint8_t scan_cycle_idx[MAX_MATRIX_KEY_COLS][MAX_MATRIX_KEY_ROWS];
+	/* Track previous "elapsed clock cycles" per matrix scan. This
+	 * is used to calculate the debouncing time for every key
+	 */
+	uint8_t scan_clk_cycle[SCAN_OCURRENCES];
+	struct k_sem poll_lock;
+	uint8_t scan_cycles_idx;
+	kscan_callback_t callback;
+	struct k_thread thread;
+	atomic_t enable_scan;
+
+	K_KERNEL_STACK_MEMBER(thread_stack, TASK_STACK_SIZE);
+};
+
+#define KSCAN_XEC_CONFIG(dev)						\
+	((struct kscan_xec_config const *)(dev)->config)
+
+#define KSCAN_XEC_DATA(dev)						\
+	((struct kscan_xec_data *const)(dev)->data)
+
+#define KSCAN_XEC_REG_BASE(dev)						\
+	((struct kscan_regs *)KSCAN_XEC_CONFIG(dev)->base_addr)
```

Comment:
please, don't use custom accessors, ref. https://github.com/zephyrproject-rtos/zephyr/pull/41918

Reply:
okay, will remove the custom accessors.

Reply:
removed custom accessors